src/ChunkMap.cpp

```diff
@@ -3041,13 +3127,16 @@ void cChunkMap::cChunkLayer::GetChunkStats(int & a_NumChunksValid, int & a_NumCh
 
 
 
-void cChunkMap::cChunkLayer::Save(void)
+void cChunkMap::cChunkLayer::Save(bool a_UnusedOnly)
 {
 	cWorld * World = m_Parent->GetWorld();
 	for (size_t i = 0; i < ARRAYCOUNT(m_Chunks); ++i)
 	{
-		if ((m_Chunks[i] != nullptr) && m_Chunks[i]->IsValid() && m_Chunks[i]->IsDirty())
+		// Queues for saving chunks that are valid, dirty, and not queued for saving.
+		// If a_UnusedOnly is true, the chunks must also be unused, in addition to the above.
+		if ((m_Chunks[i] != nullptr) && m_Chunks[i]->IsValid() && m_Chunks[i]->IsDirtyAndNotQueuedForSave() && ((!a_UnusedOnly) || m_Chunks[i]->IsUnused()))
```

Comment:
Brackets (should not be) around a_UnusedOnly.

IsDirtyAndNotQueuedForSave should again be a task for QueueSaveChunk.